test/e2e/suite-endpoints/endpoints.go

```diff
@@ -21,17 +21,27 @@ import (
 
 	ginkgo "github.com/onsi/ginkgo/v2"
 	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
 )
 
+func ensureNumListUpstreamNodes(s *scaffold.Scaffold, upsNum int, upsNodesNum int) error {
+	condFunc := func() (bool, error) {
+		ups, err := s.ListApisixUpstreams()
+		if err != nil || len(ups) != upsNum || len(ups[0].Nodes) != upsNodesNum {
```

Comment:
I thought it would be better to add some comments here.

such as expected return values, data structures and examples